dogfood/Dockerfile

```diff
@@ -312,8 +312,24 @@ COPY --from=go /tmp/bin /usr/local/bin
 COPY --from=rust-utils /tmp/bin /usr/local/bin
 COPY --from=proto /tmp/bin /usr/local/bin
 
+# Configure Nix without sandboxing - https://github.com/NixOS/nix/issues/2636#issuecomment-455302745
+RUN addgroup --system nixbld \
+  && adduser coder nixbld \
+  && for i in $(seq 1 30); do useradd -ms /bin/bash nixbld$i &&  adduser nixbld$i nixbld; done \
+  && mkdir -m 0755 /nix && chown coder /nix \
+  && mkdir -p /etc/nix && echo 'sandbox = false' > /etc/nix/nix.conf
+
 USER coder
 
+# Install Nix
+RUN sh <(curl -L https://nixos.org/nix/install) --no-daemon
+
+# Install Nix into our bash profile so `nix-shell`, `nix-build, and `nix` are available
+RUN echo '. /home/coder/.nix-profile/etc/profile.d/nix.sh' >> /home/coder/.bashrc
```

Comment:
Will this actually work? Won't this be mounted over by the users home volume?

I'd venture we can create `/nix` in the Dockerfile, but we'd have to defer nix installation to `startup_script` of the agent so that we're creating the appropriate files in the users home volume?

We might also want to modify `main.tf` to add a second user volume at `/nix`. This will allow the nix repo to be cached between workspace restarts (i.e. faster startup).

Reply:
> Will this actually work? Won't this be mounted over by the users home volume?

If overlaid, then no it won't. Nicely spotted.

> I'd venture we can create /nix in the Dockerfile, but we'd have to defer nix installation to startup_script of the agent so that we're creating the appropriate files in the users home volume?

Agreed.

> We might also want to modify main.tf to add a second user volume at /nix. This will allow the nix repo to be cached between workspace restarts (i.e. faster startup).

Solid suggestion, indeed it would.

Reply:
Pushed some changes that in theory, resolve the above. What's the easiest way to dogfood, dogfood?